### DISCUSSION TITLE
2 Bytes for Text?

### DISCUSSION THREAD

**Discussion by Flundi** [pixelit-project/PixelIt#375] [Help]
Hello,
i dont know, whats my Problem is...

`- id: '1728585596906'
  alias: Text an PixelIt senden
  triggers:
  - entity_id: input_text.pixelit_text
    trigger: state
  actions:
  - data:
      topic: pixelit/setScreen
      payload: "{\"text\": {\"textString\": \" {{ states('input_text.pixelit_text')
        }}\",\n \"bigFont\": true,\n \"scrollText\": \"auto\",\n \"scrollTextDelay\":
        40,\n \"centerText\": false,\n \"position\": {\"x\": 8, \"y\": 1},\n \"color\":
        {\"r\": 255, \"g\": 255, \"b\": 255},\n \"hexColor\": \"#FFFFFF\",   \n\n
        \"fontSize\": 2}}\n"
    action: mqtt.publish`

`# PixelIT
rest_command:
  send_pixelit_text:
    url: 'http://192.168.1.104/api/text'
    method: post
    headers:
      Content-Type: 'application/json'
    payload: >
      {
        "text": {
          "textString": "{{ input_text.pixelit_text }}",
          "bigFont": false,
          "scrollText": "auto",
          "scrollTextDelay": 20,
          "centerText": false,
          "position": {
            "x": 8,
            "y": 1
          },
          "color": {
            "r": 255,
            "g": 255,
            "b": 255
          },
          "hexColor": "#FFFFFF"
        }
      }`

Log fromPixelIT:

MQTT_callback: JSON message empty or too long
MQTT_callback: Incoming JSON (Topic: pixelit/setScreen, Cmd: setScreen, Bytes: 240/2)

**Comment by foorschtbar** [comment]
Its invalid json:

```
Error: Parse error on line 1:
...xt": {"textString": " {{ states('input_t
-----------------------^
Expecting 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '[', got 'undefined'
```

https://duckduckgo.com/?q=json+validator
